packages/components/src/autocomplete/index.tsx

```diff
@@ -163,20 +194,36 @@ export function useAutocomplete( {
 		) {
 			return;
 		}
+
 		switch ( event.key ) {
-			case 'ArrowUp':
-				setSelectedIndex(
+			case 'ArrowUp': {
+				const newIndex =
 					( selectedIndex === 0
 						? filteredOptions.length
-						: selectedIndex ) - 1
-				);
+						: selectedIndex ) - 1;
+				setSelectedIndex( newIndex );
+				if ( isAppleOS() ) {
+					speak(
+						filteredOptions[ newIndex ].textLabel ||
+							getNodeText( filteredOptions[ newIndex ].label ),
```

Comment:
I couldn't find a place where the `textLabel` and `getNodeText( filteredOptions[ newIndex ].label )` differed. Could we rely only on `getNodeText()` and remove the `textLabel` option from the component?

Reply:
@jeryj We could try but trying to parse text out of objects I thought could be tricky. I knew this would work fine for Core but third-party usages? Thoughts?